LibreNMS/OS/Junos.php

```diff
@@ -59,7 +59,7 @@ public function discoverOS(Device $device): void
 
         preg_match('/Juniper Networks, Inc. (?<hardware>\S+) .* kernel JUNOS (?<version>[^, ]+)[, ]/', $device->sysDescr, $parsed);
         if (isset($data[2]['hrSWInstalledName'])) {
-            preg_match('/\[(.+)]/', $data[2]['hrSWInstalledName'], $parsedVersion);
+            preg_match('/^JUNOS.*\[(.+?)]/', $data[2]['hrSWInstalledName'], $parsedVersion);
```

Comment:
I think it’s better we do 
```suggestion
            preg_match('/\[([^]]+)/', $data[2]['hrSWInstalledName'], $parsedVersion);
```